fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt

```diff
@@ -60,18 +60,49 @@
   </xsl:template>
 
 
+  <!-- the following template creates an _ms field for thesis advisors -->
+  <xsl:template match="mods:mods/mods:name[(mods:role/mods:roleTerm='Thesis advisor') or 
+    (mods:role/mods:roleTerm='thesis advisor')]" mode="utk_ir_MODS">
+
+    <xsl:for-each select=".">
+      <xsl:variable name="given-n" select="mods:namePart[@type='given']"/>
+      <xsl:variable name="family-n" select="mods:namePart[@type='family']"/>
+      <xsl:variable name="t-o-address" select="mods:namePart[@type='termsOfAddress']"/>
+      <xsl:variable name="display-f" select="mods:displayForm" />
+
+
+      <field name="utk_mods_etd_thesis_advisor_ms">  
+          <xsl:choose>
+	     <xsl:when test="$display-f!=''">
```

Comment:
So, small logic error here.  If we do it this way, we can never update the value if they do it right.

Instead, switch this so:

1. test if termsOfAddress is not empty
2. test if you have a family or given
3. otherwise, displayform